detection_rules/ghwrap.py

```diff
@@ -98,6 +99,41 @@ def download_gh_asset(url: str, path: str, overwrite=False):
     z.close()
 
 
+def update_gist(token: str,
+                file_map: Dict[Path, str],
+                description: str,
+                gist_id: str,
+                public=False,
+                pre_purge=False) -> Response:
+    """Update existing gist."""
+    url = f'https://api.github.com/gists/{gist_id}'
+    headers = {
+        'accept': 'application/vnd.github.v3+json',
+        'Authorization': f'token {token}'
+    }
+    body = {
+        'description': description,
+        'files': {},  # {path.name: {'content': contents} for path, contents in file_map.items()},
+        'public': public
+    }
+
+    if pre_purge:
+        # retrieve all existing file names which are not in the file_map and overwrite them to empty to delete files
+        # only. When a file is completely deleted and updated, the raw URL changes
+        response = requests.get(url)
+        response.raise_for_status()
+        data = response.json()
+        files = list(data['files'])
+        body['files'] = {file: {} for file in files if file not in file_map}
+        response = requests.patch(url, headers=headers, json=body)
+        response.raise_for_status()
```

Comment:
I think 401 for bad creds. Do you think 404 not found is not specific enough for `a gist file that they can not write to.`. GitHub is sometimes overly generic and so we'd have to make assumptions, because it could be a bad gist ID just as much as missing, so in this case, I think we should be ok (especially for an internal `dev` command)

Reply:
Both may be good. However, I only received the 404 error when testing.

Reply:
d36a83